### PR TITLE
profile.d/go.sh: add GOPATH

### PR DESCRIPTION
**Pull request by ChickenNuggers**
`$GOPATH` (documented [here](https://github.com/golang/go/wiki/GOPATH)) is respected by Go and GoPM. Any other Go tools **should** respect this variable.

### CODE REVIEW

File: profile.d/go.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# oh my go.sh
+# https://github.com/golang/go/wiki/GOPATH
+export GOPATH="$HOME/go"
```
Comment (KellerFuchs) [inline profile.d/go.sh]: 1. Using `~/go` for Go's internal source store is kinda dirty/confusing.  Why not `~/.cache/gocode` or `~/.local/gocode`?
2. This is masking system-wide Go packages that live in `/usr/share/gocode`. Is it on purpose?

Comment (KellerFuchs) [inline profile.d/go.sh]: They do install there.

**Comment by ChickenNuggers** [comment]
@KellerFuchs should be resolved

**Comment by ChickenNuggers** [comment]
cool i'll fix merge errors then merge in a few

**Pull request merged**